src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5883,66 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+#nullable enable
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            lastTokenOfType = EatToken(SyntaxKind.AsteriskToken);
+
+            if (IsPossibleCallingConvention())
+            {
+                _ = EatToken();
+            }
+
+            if (!IsPossibleFunctionPointerParameterListStart(CurrentToken))
+            {
+                return ScanTypeFlags.MustBeType;
+            }
+
+            _ = EatToken(SyntaxKind.LessThanToken);
```

Comment:
>EatToken(SyntaxKind.LessThanToken) [](start = 16, length = 34)

Is this going to eat ```OpenParenToken``` for which ```IsPossibleFunctionPointerParameterListStart(CurrentToken)``` above returns true?

Reply:
No, I should just make this a `EatToken()`

---
In reply to: [349366215](https://github.com/dotnet/roslyn/pull/39879#discussion_r349366215) [](ancestors = 349366215)

Reply:
And same with the `EatToken(SyntaxKind.GreatherThanToken)` below

---
In reply to: [349367034](https://github.com/dotnet/roslyn/pull/39879#discussion_r349367034) [](ancestors = 349367034,349366215)